Source/WebKitLegacy/WebCoreSupport/WebResourceLoadScheduler.h

```diff
@@ -83,8 +83,10 @@ class WebResourceLoadScheduler final : public WebCore::LoaderStrategy {
     bool isSuspendingPendingRequests() const { return !!m_suspendPendingRequestsCount; }
     void isResourceLoadFinished(WebCore::CachedResource&, CompletionHandler<void(bool)>&&) final;
 
-    class HostInformation : public CanMakeWeakPtr<HostInformation>, public CanMakeCheckedPtr<HostInformation> {
-        WTF_MAKE_NONCOPYABLE(HostInformation); WTF_MAKE_FAST_ALLOCATED;
+    class HostInformation final : public CanMakeWeakPtr<HostInformation>, public CanMakeCheckedPtr<HostInformation> {
```

Comment:
Shouldn't this be `CanMakeCheckedPtr` instead of `CanMakeCheckedPtr<HostInformation>`? Same comment may apply elsewhere.

Reply:
CanMakeCheckedPtr requires the subclass type as a template parameter in order to verify that the type includes the WTF_OVERRIDE_DELETE_FOR_CHECKED_PTR macro. 

Is there some fancy C++ I need to learn? How would "CanMakeCheckedPtr" infer the type parameter?